MediaProcessor/src/FFmpegCommandBuilder.cpp

```diff
@@ -6,71 +6,139 @@
+FFmpegCommandBuilder& FFmpegCommandBuilder::addOutputFile() {
+    addArgument(m_ffmpegConfig.getOutputFilePath());
 
-    addFlag("-c:a", codec);
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addAudioCodec() {
+    addFlag("-c:a", Utils::enumToString<AudioCodec>(m_ffmpegConfig.getAudioCodec(),
+                                                    m_ffmpegConfig.getAudioCodecAsString()));
 
     return *this;
 }
 
 FFmpegCommandBuilder& FFmpegCommandBuilder::addAudioSampleRate() {
-    int sampleRate = m_ffmpegConfig.getAudioSampleRate();
-    addFlag("-ar", std::to_string(sampleRate));
+    addFlag("-ar", std::to_string(m_ffmpegConfig.getAudioSampleRate()));
 
     return *this;
 }
 
 FFmpegCommandBuilder& FFmpegCommandBuilder::addAudioChannels() {
-    int channels = m_ffmpegConfig.getAudioChannels();
-    addFlag("-ac", std::to_string(channels));
+    addFlag("-ac", std::to_string(m_ffmpegConfig.getAudioChannels()));
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addStartTime(const std::ostringstream& startTime) {
+    addFlag("-ss", startTime.str());
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addDuration(const std::ostringstream& duration) {
+    addFlag("-t", duration.str());
 
     return *this;
 }
 
 FFmpegCommandBuilder& FFmpegCommandBuilder::addVideoCodec() {
-    std::string codec = Utils::enumToString<VideoCodec>(m_ffmpegConfig.getVideoCodec(),
-                                                        m_ffmpegConfig.getVideoCodecAsString());
+    addFlag("-c:v", Utils::enumToString<VideoCodec>(m_ffmpegConfig.getVideoCodec(),
+                                                    m_ffmpegConfig.getVideoCodecAsString()));
+
+    return *this;
+}
 
-    addFlag("-c:v", codec);
+FFmpegCommandBuilder& FFmpegCommandBuilder::addChunkPath() {
+    addArgument(m_ffmpegConfig.getChunksPath());
 
     return *this;
 }
 
-std::string FFmpegCommandBuilder::build() const {
-    if (m_inputFilePath.empty()) {
-        throw std::runtime_error("Input file path must be specified.");
-    }
+FFmpegCommandBuilder& FFmpegCommandBuilder::addStrictExperimental() {
+    addFlag("-strict", "experimental");
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addStream(const std::string map) {
+    addFlag("-map", map);
+
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addShortest() {
+    addFlag("-shortest");
 
-    if (m_outputFilePath.empty()) {
-        throw std::runtime_error("Output file path must be specified.");
+    return *this;
+}
+
+FFmpegCommandBuilder& FFmpegCommandBuilder::addFilterComplex(
+    const std::vector<fs::path> processedChunkCol) {
+    addFlag("-filter_complex", buildFilterComplex(processedChunkCol));
+
+    return *this;
+}
+
+std::string FFmpegCommandBuilder::buildCrossfade(int chunkColSize) const {
+    std::string filterComplex = "";
```

Comment:
not that important but let's remove the `= ""` as the default string ctor does that for us. Also, good call on moving this guy here.